### PR TITLE
Add Montclair EDFA Support

### PR DESCRIPTION
**Pull request by peelman**
Adds support for Montclair's [EDFA amplifiers](https://montclairfiber.com/product/edfa-high-power-optical-amplifier/): 

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [X] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by peelman** [comment]
Tests are failing due to Valere OSes....this didn't touch Valere OS files...anybody help here?  @Jellyfrog @murrant ?

**Comment by Jellyfrog** [comment]
I'm guessing Because the oid you use to detect it collides with the other OS

**Comment by peelman** [comment]
It doesn't collide, but it does overlap.  Valeres are 13858.  Montclair is using 1385.  There another bad regex somewhere?

File: mibs/montclair/Montclair-MIB.mib [hunk 1]
```diff
@@ -0,0 +1,949 @@
+Montclair-MIB DEFINITIONS ::= BEGIN
```
Comment (Jellyfrog) [inline mibs/montclair/Montclair-MIB.mib:1]: rename this file to Montclair-MIB

File: includes/definitions/montclair-edfa.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+mib_dir: montclair
+icon: montclair
+over:
+    - { graph: device_dbm, text: 'Optical Power' }
+    - { graph: device_temperature, text: 'Device Temperature' }
+
+discovery:
+    -
+        sysObjectID:
+            - .1.3.6.1.4.1.1385
```
Comment (Jellyfrog) [inline includes/definitions/montclair-edfa.yaml:15]: Try
```suggestion
            - .1.3.6.1.4.1.1385.
```

Comment (peelman) [reply includes/definitions/montclair-edfa.yaml]: Well...that seems to have led to some wildly new an interesting errors...

File: LibreNMS/OS/MontclairEdfa.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+use App\Models\Device;
+use LibreNMS\Interfaces\Discovery\OSDiscovery;
+use LibreNMS\OS;
+
+class MontclairEdfa extends OS implements OSDiscovery
+{
+    public function discoverOS(Device $device): void
+    {
+        $data = snmp_get_multi($this->getDeviceArray(), [
+            'Montclair-MIB::devicePartNumber.0',
```
Comment (murrant) [inline LibreNMS/OS/MontclairEdfa.php:37]: It is better to use yaml for these now.

Comment (peelman) [reply LibreNMS/OS/MontclairEdfa.php]: Done.

**Comment by peelman** [comment]
Pulled in latest changes from `master`.

**Review by murrant — APPROVED** [review]

**Pull request merged**